server/property-routes.ts

```diff
@@ -155,12 +166,32 @@ router.put('/:propertyId', async (req, res, next) => {
     console.log('Put Property Request params:', req.params);
     console.log('Put Property Request body:', req.body);
 
+    // Verify user is authenticated
+    const userId = Number(req.user?.userId);
+    if (!userId) {
+      throw new ClientError(401, 'Authentication required');
+    }
+
     const propertyId = Number(req.params.propertyId);
     if (!propertyId) {
       console.warn('Invalid propertyId provided:', req.params.propertyId);
       throw new ClientError(400, 'propertyId is required');
     }
 
+    // Verify Owner
+    const verifyOwnerSql = `
```

Comment:
This is not necessary and it requires an extra database query, which can get expensive. Instead, you simply add the `"userId" = $2` to the where clause in the update statement.